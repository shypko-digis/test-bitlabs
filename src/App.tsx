import React from 'react';
import Modal from './containers/modal';

import './App.css';

function App() {
  return (
    <div className='App'>
      <Modal title='Select Your Amount' />
    </div>
  );
}

export default App;
