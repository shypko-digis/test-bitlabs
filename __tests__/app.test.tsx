import React from 'react';
// import * as enzyme from 'enzyme';
import { shallow, configure } from 'enzyme';
// import * as Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

import App from '../src/App';
import Modal from './../src/containers/modal/index';
import ReactSixteenAdapter from 'enzyme-adapter-react-16';

configure({ adapter: new ReactSixteenAdapter() });

it('renders correctly', () => {
  renderer.create(<App />);
});

it('snapshot modal', () => {
  const wrapper = shallow(<Modal title='title' />);
  expect(wrapper).toMatchSnapshot();
});


// it('is exist input usd', () => {
//   const wrapper = shallow(<Modal title='aaaaa' />);
//   expect(wrapper.find('#usd')).toExist();
// });

it('check input 1', () => {
  const wrapper = shallow(<Modal title='title' />);
  wrapper.find("#usd").simulate('change', { target: { value: 200 } })
  expect(wrapper.find("#usd").get(0).props.value).toEqual(200);
  expect(wrapper.find("#crypta").get(0).props.value).toBeGreaterThan(0)

});

it('check input 2', () => {
  const wrapper = shallow(<Modal title='title' />);
  wrapper.find("#crypta").simulate('change', { target: { value: 200 } })
  expect(wrapper.find("#crypta").get(0).props.value).toEqual(200);
  expect(wrapper.find("#usd").get(0).props.value).toBeGreaterThan(0)

});

it('check input err', () => {
  const wrapper = shallow(<Modal title='title' />);
  wrapper.find("#crypta").simulate('change', { target: { value: 0 } })
  expect(wrapper.find("#crypta").get(0).props.value).toEqual(0);
  expect(wrapper.find("#usd").get(0).props.value).toBeGreaterThan(0)
});