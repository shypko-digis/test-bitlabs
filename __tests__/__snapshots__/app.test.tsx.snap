// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot modal 1`] = `ShallowWrapper {}`;
